src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs

```diff
@@ -138,116 +134,64 @@ public void InitializeService(ServiceHost serviceHost)
             return Task.CompletedTask;
         }
 
-        /// <summary>
-        /// Generates the contextualization scripts for a server. The generated context is in the form of create scripts for
-        /// database objects like tables and views.
-        /// </summary>
-        /// <param name="contextualizationParams">The contextualization parameters.</param>
-        internal static async Task GenerateServerContextualization(GenerateServerContextualizationParams contextualizationParams, RequestContext<GenerateServerContextualizationResult> requestContext)
+        internal static async Task GetServerContextualization(GetServerContextualizationParams contextualizationParams, RequestContext<GetServerContextualizationResult> requestContext)
         {
             MetadataService.ConnectionServiceInstance.TryFindConnection(contextualizationParams.OwnerUri, out ConnectionInfo connectionInfo);
-
-            if (connectionInfo != null)
+            if (connectionInfo == null)
+            {
+                Logger.Error("Failed to find connection info about the server.");
+                await requestContext.SendError("Failed to find connection info about the server.");
```

Comment:
I'd suggest just throwing an exception. That way you're guaranteeing it won't keep executing and makes it clear that that's the end of execution

(STS will catch exceptions and send them as errors by default)